app/main/views/suppliers.py

```diff
@@ -313,12 +315,50 @@ def deactivate_user(user_id):
     return redirect(url_for('.find_supplier_users', supplier_id=user['users']['supplier']['supplierId']))
 
 
-@main.route('/suppliers/users/<int:user_id>/remove-from-supplier', methods=['POST'])
+@main.route('/suppliers/<int:supplier_id>/move-existing-user', methods=['POST'])
 @login_required
 @role_required('admin')
-def remove_from_supplier(user_id):
-    data_api_client.update_user(user_id, role='buyer', updater=current_user.email_address)
-    return redirect(url_for('.find_supplier_users', supplier_id=request.form['supplier_id']))
+def move_user_to_new_supplier(supplier_id):
+    form = EmailAddressForm()
+    form2 = MoveUserForm()
+
+    try:
+        suppliers = data_api_client.get_supplier(supplier_id)
+        users = data_api_client.find_users(supplier_id)
+    except HTTPError as e:
+        current_app.logger.error(str(e), supplier_id)
+        if e.status_code != 404:
+            raise
+        else:
+            abort(404, "Supplier not found")
+
+    if form2.validate_on_submit():
+        try:
+            user = data_api_client.get_user(email_address=form2.user_to_move_email_address.data)
+        except HTTPError as e:
+            current_app.logger.error(str(e), supplier_id)
+            raise
+
+        if user:
+            data_api_client.update_user(
+                user['users']['id'],
+                role='supplier',
+                supplier_id=supplier_id,
+                active=True,
+                updater=current_user.email_address
+            )
+            flash("user_moved", "success")
+        else:
+            flash("user_not_moved", "error")
+        return redirect(url_for('.find_supplier_users', supplier_id=supplier_id))
+    else:
+        return render_template(
+            "view_supplier_users.html",
+            form=form,
```

Comment:
I don't know if we ever do this, but if you just initialized the form here (ie, `form=EmailAddressForm()`), then it would be clear that we don't use it in this route at all.  
By initializing it at the beginning, it seems like it might be being used in the view somewhere.